### DISCUSSION TITLE
Is it possible to remove users from a group?

### DISCUSSION THREAD

**Discussion by blacksabbatherMJ** [gram-js/gramjs#257] [General]
I already figured out using iterParticipants to list all participants from a particular group. How to remove participants from a group? Which function can I call? Thanks~

**Comment by rojvv** [comment]
Try this for supergroups:

```ts
await client.invoke(
  new Api.channels.EditBanned({
    channel: channelId,
    participant: userId,
    bannedRights: new Api.ChatBannedRights({
      untilDate: 0,
      viewMessages: true,
      sendMessages: true,
      sendMedia: true,
      sendStickers: true,
      sendGifs: true,
      sendGames: true,
      sendInline: true,
      embedLinks: true
  })
})
```

And this for small groups:

```ts
await client.invoke(
  new Api.messages.DeleteChatUser({
    chatId: chatId,
    userId: userId
  })
)
```

**Reply by blacksabbatherMJ** [reply]
My target is a channel. If I use Api.channels.EditBanned, will that user still be displayed and counted in the total user number? Thanks~

**Reply by blacksabbatherMJ** [reply]
The background is that I can always find a bunch of 'Deleted Account' which I want to remove from my channel. And I want to do it programmatically

**Reply by rojvv** [reply]
> My target is a channel. If I use Api.channels.EditBanned, will that user still be displayed and counted in the total user number? Thanks~

If you use it like I showcased, no it won't be displayed.

**Reply by rojvv** [reply]
@blacksabbatherMJ could you do what you wanted to?

**Comment by blacksabbatherMJ** [comment]
Guys,

I'm afraid I have to re-phrase my questions as TG is so confusing:
**Question 1 - How to list all users from a particular group**
I used below code:
```ts
const participants = client.iterParticipants("myGroup", {
   limit: 10000,
});
```
and participants only gives me 750 users meanwhile in TG app, it shows I have around 2000 users.

**Question 2 - How to remove all "Deleted Account":**
The whole background is in our group I can find quite a few users displayed as "Deleted Account" which I'd like to remove programatically. How to list all of them? I used
```ts
if(participant.deleted){
   console.log("username text is : ", participant.username);
}
```
And it also gives mis-matched number...
So I need a whole solution to list all "Deleted Account" and remove them from my group

**Reply by painor** [reply]
if they are deleted they wouldn't have a username,  you probably wanna log their IDs instead,

If it's a channel (not a supergroup) you will only be able to fetch 200 users not all of them, it's a telegram limit.

**Reply by blacksabbatherMJ** [reply]
Not true. I can get 750 users but my channel has more than that
